File: ShopUser/src/app/main/user/model/request/sign-up.request.ts
import {SignUpModel} from "../sign-up.model";

export class SignUpRequest {
  constructor(
    public readonly email: string,
    public readonly password: string,
    public readonly passwordConfirm: string,
    public readonly userName: string,
    public readonly mobile: string,
    public readonly postCode: string,
    public readonly address: string,
    public readonly extraAddress: string
  ) { }

  public static mapFrom(model: SignUpModel) {
    return new SignUpRequest(
      model.email,
      model.password,
      model.passwordConfirm,
      model.userName,
      model.mobile,
      model.postCode,
      model.address,
      model.extraAddress);
  }
}

File: ShopAdmin/src/main/java/com/vision/shoppingmall/user/model/request/SignUpRequest.java
package com.vision.shoppingmall.user.model.request;

import lombok.Getter;

@Getter
public class SignUpRequest {
  private String email;
  private String password;
  private String passwordConfirm;
  private String userName;
  private String mobile;
  private String postCode;
  private String address;
  private String extraAddress;
}
